### ISSUE TITLE
reping k should be -1

### ISSUE THREAD

**Issue by BiuBiuBiLL** [hlin117/mdlp-discretization#3]
```
cdef SIZE_t find_cut(np.ndarray[np.int64_t, ndim=1] y, int start, int end):
    """Finds the best cut between the specified interval. The cut returned is
    an index k. If k split the array into two sub arrays A and B, then
    k is the last index of A. In other words, let start == 0 and end == n.
    We might have the array
        [0, 1, 2, 3, ..., k - 1] [k, k + 2, ..., n]
    Therefore, k is in the range {1, 2, ..., n}. (If k == n, then array B
    is simply of length 1.)
    If k = -1, then no good cut point was found.
```

In the code explanation, you already said that if k=-1, then no good cut point was found. But in your code, there is no path lead to k=-1.

**Comment by hlin117** [comment]
I'll look into this issue, thanks.

**Comment by hlin117** [comment]
Closed by commit 886b78b8948dbfa1c37b05cc354ecbb8ccbe1bd3. Thanks for the find.
